dbms/src/Storages/DeltaMerge/tests/gtest_dm_minmax_index.cpp

```diff
@@ -157,170 +157,1087 @@ bool checkPkMatch(const String & test_case, Context & context, const String & ty
-    ASSERT_EQ(false, checkMatch(case_name, *context, "MyDateTime", "2020-09-27", createIn(attr("MyDateTime"), {parseMyDateTime("2020-09-28")})));
-    ASSERT_EQ(true, checkMatch(case_name, *context, "MyDateTime", "2020-09-27", createGreater(attr("MyDateTime"), parseMyDateTime("2020-09-26"), 0)));
-    ASSERT_EQ(false, checkMatch(case_name, *context, "MyDateTime", "2020-09-27", createGreater(attr("MyDateTime"), parseMyDateTime("2020-09-27"), 0)));
-    ASSERT_EQ(true, checkMatch(case_name, *context, "MyDateTime", "2020-09-27", createGreaterEqual(attr("MyDateTime"), parseMyDateTime("2020-09-27"), 0)));
-    ASSERT_EQ(false, checkMatch(case_name, *context, "MyDateTime", "2020-09-27", createGreaterEqual(attr("MyDateTime"), parseMyDateTime("2020-09-28"), 0)));
-    ASSERT_EQ(true, checkMatch(case_name, *context, "MyDateTime", "2020-09-27", createLess(attr("MyDateTime"), parseMyDateTime("2020-09-28"), 0)));
-    ASSERT_EQ(false, checkMatch(case_name, *context, "MyDateTime", "2020-09-27", createLess(attr("MyDateTime"), parseMyDateTime("2020-09-27"), 0)));
-    ASSERT_EQ(true, checkMatch(case_name, *context, "MyDateTime", "2020-09-27", createLessEqual(attr("MyDateTime"), parseMyDateTime("2020-09-27"), 0)));
-    ASSERT_EQ(false, checkMatch(case_name, *context, "MyDateTime", "2020-09-27", createLessEqual(attr("MyDateTime"), parseMyDateTime("2020-09-26"), 0)));
-
-    ASSERT_EQ(false, checkDelMatch(case_name, *context, "Int64", "100", createEqual(attr("Int64"), Field((Int64)100))));
-    ASSERT_EQ(true, checkPkMatch(case_name, *context, "Int64", "100", createEqual(pkAttr(), Field((Int64)100)), true));
-    ASSERT_EQ(true, checkPkMatch(case_name, *context, "Int64", "100", createGreater(pkAttr(), Field((Int64)99), 0), true));
-    ASSERT_EQ(true, checkPkMatch(case_name, *context, "Int64", "100", createGreater(pkAttr(), Field((Int64)99), 0), false));
-
-    ASSERT_EQ(true, checkMatch(case_name, *context, "Int64", "100", createNotEqual(attr("Int64"), Field((Int64)101))));
-    ASSERT_EQ(true, checkMatch(case_name, *context, "String", "test_like_filter", createLike(attr("String"), Field(Field((String) "*filter")))));
-    ASSERT_EQ(true, checkMatch(case_name, *context, "String", "test_not_like_filter", createNotLike(attr("String"), Field(Field((String) "*test_like_filter")))));
-    ASSERT_EQ(true, checkMatch(case_name, *context, "Int64", "100", createNotIn(attr("Int64"), {Field((Int64)101), Field((Int64)102), Field((Int64)103)})));
-    ASSERT_EQ(true, checkMatch(case_name, *context, "Int64", "100", createIn(attr("Int64"), {Field((Int64)100), Field((Int64)101), Field((Int64)102)})));
-    // clang-format on
+
+    for (size_t operater_type = Test_Equal; operater_type < Test_MaxOperator; operater_type++)
+    {
+        for (size_t datatype = Test_Int64; datatype < Test_Decimal64; datatype++)
+        {
+            {
+                // has null
+                if (!IsNullableDateType(static_cast<MinMaxTestDatatype>(datatype)))
```

Comment:
> Seems it will return early for non nullable type, and no test will be run for these types.

Because some values in this branch will have `Null` value, so data types should be Nullable.

Reply:
I mean it seems there is not test for non-nullable type.

Reply:
> 

In Line 1065 will test all data types, including non-nullable types.